django/views/generic/base.py

```diff
@@ -27,8 +27,12 @@ def get_context_data(self, **kwargs):
         kwargs.setdefault('view', self)
         if self.extra_context is not None:
             kwargs.update(self.extra_context)
+        kwargs.update(self.get_context_updates())
         return kwargs
 
+    def get_context_updates(self):
```

Comment:
what are the benefits of this? can you also provide tests to verify this changes?

Reply:
Thanks for the response! This is not something that will change the way django works, it is just a little patch to save developers from writing unnecessary 2 lines of code every time they want to dynamically update context

Reply:
it would be better to add test to demonstrate that